teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -504,156 +705,74 @@ Cons:
-metadata:
-  name: run
-spec:
-  pipelineRef:
-    name: build-test-deply
-  workspaces:
-  - name: source-code
-    emptyDir: {}
-  - name: test-results
-    emptyDir: {}
+    - name: source-code
+    - name: test-results
 ```
 
-Running with this PipelineRun would cause all of the Tasks to be run in separate pods:
-
-```yaml
-apiVersion: tekton.dev/v1beta1
-kind: PipelineRun
-metadata:
-  name: run
-spec:
-  pipelineRef:
-    name: build-test-deply
-  workspaces:
-  - name: source-code
-    persistentVolumeClaim:
-      claimName: otherpvc
-  - name: test-results
-    persistentVolumeClaim:
-      claimName: mypvc
-```
+Workspace init and teardown could be done either for every Task (which could be quite a bit of overhead) or once for the
+Pipeline.
 
 Pros:
-* Allows runtime decisions about scheduling without changing the Pod
+* Solves one specific problem: getting data on and off of workspaces
 
 Cons:
-* If it's important for a Pipeline to be executed in a certain way, that information will have to be encoded somewhere
-  other than the Pipeline 
-* For very large Pipelines, this default behavior may cause problems (e.g. if the Pipeline is too large to be scheduled
-  into one pod)
-* A bit strange and confusing to overload the meaning of `emptyDir`, might be simpler and clearer to have a field instead
-
-#### PipelineRun: field
-
-This is similar to the `emptyDir` based solution but instead of adding extra meaning to `emptyDir` we add a field to the
-runtime workspace information or to the entire PipelineRun (maybe when this field is set workspaces do not need to be
-provided.)
-
-A field could also be added as part of the Pipeline definition if desired (vs at runtime via a PipelineRun).
-
-#### Controller level
-
-This option is [TEP-0046](https://github.com/tektoncd/community/pull/318). In this option, the Tekton controller can
-be configured to always execute Pipelines inside one pod.
-
-Pros:
-* Authors of PipelineRuns and Pipelines don't have to think about how the Pipeline will be executed
-* Pipelines can be used without updates
+* Helps with our conceptual problems but not with our efficiency problems
+* The difference between a workspace teardown and finally could be confusing
 
-Cons:
-* Only cluster administrators will be able to control this scheduling, there will be no runtime or authoring time
-  flexibility
-* Executing a pipeline in a pod will require significantly re-architecting our graph logic so it can execute outside
-  the controller and has a lot of gotchas we'll need to iron out (see
-  [https://hackmd.io/@vdemeester/SkPFtAQXd](https://hackmd.io/@vdemeester/SkPFtAQXd) for some more brainstorming)
+Related:
+* [Task Specialization: most appealing options?](https://docs.google.com/presentation/d/12QPKFTHBZKMFbgpOoX6o1--HyGqjjNJ7own6KqM-s68)
 
-### Within the Task
+##### Within the Task
 
 In this option we ignore the "not at Task authoring time" requirement and we allow for Tasks to contain other Tasks.
```

Comment:
Is there a requirement that colocation isn't specified at authoring time? This line is confusing as I don't see something similar in the requirements section and many of the proposed alternatives are at authoring time.